### PR TITLE
[Almost Done] Clique POA/Rinkeby Read Support

### PR DESCRIPTION
**Pull request by thefallentree**
#5687 
do not merge yet, this is just for testing.

### CODE REVIEW

**Comment by naikmyeong** [comment]
Can we add rinkeby chain config here? @thefallentree https://github.com/ethereum/go-ethereum/blob/master/params/config.go#L83

**Comment by thefallentree** [comment]
> Can we add rinkeby chain config here? @thefallentree https://github.com/ethereum/go-ethereum/blob/master/params/config.go#L83

yeah, on my todo list

**Comment by thefallentree** [comment]
Status Update:
1.  join https://gitter.im/goerli/clique for all discussions. 
2. latest Code sync with goreli testnet fine, still working on sync issues with RInkeby.

**Comment by thefallentree** [comment]
Status update: The latest code sync to Rinkeby without issue now.

Now, remaining work are basically just code quality improvements.

File: parity/params.rs [hunk 1]
```diff
@@ -45,6 +45,7 @@ pub enum SpecType {
 	Morden,
 	Ropsten,
 	Kovan,
+    Rinkeby,
```
Comment (5chdn) [inline parity/params.rs]: Please use tabs here

File: parity/params.rs [hunk 2]
```diff
@@ -125,6 +128,7 @@ impl SpecType {
 			SpecType::Morden => Ok(ethereum::new_morden(params)),
 			SpecType::Ropsten => Ok(ethereum::new_ropsten(params)),
 			SpecType::Kovan => Ok(ethereum::new_kovan(params)),
+            SpecType::Rinkeby => Ok(ethereum::new_rinkeby(params)),
```
Comment (5chdn) [inline parity/params.rs]: tabs

File: parity/params.rs [hunk 3]
```diff
@@ -75,6 +76,7 @@ impl str::FromStr for SpecType {
 			"morden" | "classic-testnet" => SpecType::Morden,
 			"ropsten" => SpecType::Ropsten,
 			"kovan" | "testnet" => SpecType::Kovan,
+            "rinkeby" => SpecType::Rinkeby,
```
Comment (5chdn) [inline parity/params.rs]: tabs

File: ethcore/res/ethereum/rinkeby.json [hunk 4]
```diff
@@ -0,0 +1,315 @@
+{
+	"name": "Rinkeby",
+	"dataDir": "test",
```
Comment (5chdn) [inline ethcore/res/ethereum/rinkeby.json]: should be `rinkeby`

File: ethcore/res/ethereum/rinkeby.json [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+		"clique": {
+			"params": {
+				"period": 15,
+				"epoch": 30000
+			}
+		}
+	},
+	"params": {
+		"gasLimitBoundDivisor": "0x0400",
+		"registrar": "0xe7410170f87102df0055eb195163a03b7f2bff4a",
```
Comment (5chdn) [inline ethcore/res/ethereum/rinkeby.json]: there is no registrar on rinkeby yet, can be removed

**Comment by 5chdn** [comment]
Can you add Görli, too?

https://github.com/goerli/testnet/blob/master/parity/goerli.json

**Comment by 5chdn** [comment]
Merged into https://github.com/goerli/parity-goerli for easier collaboration. Creating a new PR once closer to be reviewable.

**Pull request closed without merging**